glossary.md

```diff
@@ -0,0 +1,172 @@
+_Module 6_
+
+**MTTR** - "Mean Time To Recovery", how long it takes to fix critical system failures.
+DevOps principals encourage aiming to reduce MTTR over increasing MTBF as a better route to moving quickly with low downtime.  
+_Module 6_
+
+**Penetration testing** - AKA Pen testing. Evaluating the security of a system by attempting to breach it.  
+_Module 10_
+
+**Patching** - Applying "patches" (updates) to a server/service, which might be to the operating system, components like web servers or direct library dependencies.
+It's important to apply security patches quickly when they become available.
+CD, Immutable infrastructure and containerisation can all help make easier.
+Tools like Dependabot or Renovate can automate raising PRs to update dependencies.  
+_Module 5_
+
+**Pipeline** - An automated CI/CD process. The pipeline analogy comes from having several steps run in sequence, like passing through a pipe.  
+_Module 7_
+
+**Post-Mortem** - A review session after an incident to determine what went wrong and how to prevent a repeat.
+Sometimes called a retro or retrospective.  
+_Module 6_
+
+**Refactoring** - Improving the layout of code without changing behaviour.  
+_Module 2_
+
+**Refreshing** (a server) - Refreshing or recycling a server refers to spinning up a replacement server when using immutable infrastructure.  
+_Module 5_
+
+**Retrospective** - A review session to reflect and suggest improvements to your processes.  
+_Module 6_
+
+**SaaS** - "Software as a Service" software that you pay for, often per-user, and don't have to maintain.
+This is in comparison to enterprise software you might purchase and host yourself, or bespoke software that you write and host yourself.  
+_Module 11_
+
+**Test double** - When writing tests you might want to exclude some code, or an external dependency, to make your tests faster to write and run.
+A few types of test double are (these terms are often used interchangeably in practice):
+
+* Stub - a stub implementation is a test double that returns fixed responses to calls
+* Fake - a implementation that behaves a bit like the real thing, for example, using a hash table instead of a Redis instance
+* Mock - similar to a stub, but with some extra tricks, like recording (spying on) calls to it so that you can assert that a specific function was called
+
+_Module 3_
+
+**TDD** - "Test Driven Development", the practice of writing failing tests before writing code.  
+_Module 3_
+
+**Test Pyramid** - A diagram showing the idea of having fewer higher level end-to-end tests built on many lower level unit tests.  
+_Module 3_
+
+**UAT** - "User Agent Testing" manual testing performed by the end user to determine whether a system satisfies the requirements. Also known as Acceptance Testing.  
```

Comment:
Isn't this usually "User Acceptance Testing"? Or is that what the final sentence is implying?